src/number_theory/bernoulli.lean

```diff
@@ -139,8 +163,48 @@ begin
   rw ← cast_sub hk,
   congr',
   field_simp [show ((n - k : ℕ) : ℚ) + 1 ≠ 0, by {norm_cast, simp}],
-  -- down to nat
   norm_cast,
   rw [mul_comm, nat.sub_add_eq_add_sub hk],
   exact choose_mul_succ_eq n k,
 end
+
+open power_series
+open nat
+
+@[simp] lemma constant_coeff_exp : constant_coeff ℚ (exp ℚ) = 1 := rfl
+
+theorem algebra_map_id (q : ℚ) : algebra_map ℚ ℚ q = q :=
+begin
+  rw show algebra_map ℚ ℚ = (by refine_struct { to_fun := id}; tidy),
+  by simp only [eq_iff_true_of_subsingleton], refl,
```

Comment:
```suggestion
  rw [show algebra_map ℚ ℚ = (by refine_struct { to_fun := id}; tidy),
    by simp only [eq_iff_true_of_subsingleton]], refl,
```

not necessary but I feel like it's more readable? Right now it looks like the `by` shouldn't be there but it's necessary.

Reply:
Done!